libpod/runtime.go

```diff
@@ -706,19 +706,32 @@ func (r *Runtime) TmpDir() (string, error) {
 	return r.config.Engine.TmpDir, nil
 }
 
-// GetConfig returns a copy of the configuration used by the runtime
-func (r *Runtime) GetConfig() (*config.Config, error) {
+// GetConfig returns the configuration used by the runtime.
+// Note that the returned value is not a copy and must hence
+// only be used in a reading fashion.
+func (r *Runtime) GetConfigNoCopy() (*config.Config, error) {
```

Comment:
I'm a little uncomfortable about this one, since it does give a backdoor into the Libpod config. I am OK with merging but would prefer more warnings telling people to avoid whenever possible, and how this is fastpath only, etc.

Reply:
(If we could get lint to throw errors whenever someone adds a use, requiring a manual whitelisting, that would be neat, but that sounds beyond the scope of this PR)